src/usr/local/www/interfaces_gre_edit.php

```diff
@@ -175,32 +227,66 @@ function build_parent_list() {
 	$pconfig['remote-addr']
 ))->setHelp('Peer address where encapsulated GRE packets will be sent.');
 
-$section->addInput(new Form_IpAddress(
+$group = new Form_Group(gettext('IPv4'));
+
+$group->add(new Form_IpAddress(
 	'tunnel-local-addr',
 	'*GRE tunnel local address',
 	$pconfig['tunnel-local-addr']
 ))->setHelp('Local GRE tunnel endpoint.');
 
-$section->addInput(new Form_IpAddress(
+$group->add(new Form_IpAddress(
 	'tunnel-remote-addr',
 	'*GRE tunnel remote address',
 	$pconfig['tunnel-remote-addr']
 ))->setHelp('Remote GRE address endpoint.');
 
-$section->addInput(new Form_Select(
+$group->add(new Form_Select(
 	'tunnel-remote-net',
 	'*GRE tunnel subnet',
 	$pconfig['tunnel-remote-net'],
-	array_combine(range(128, 1, -1), range(128, 1, -1))
+	array_combine(range(32, 1, -1), range(32, 1, -1))
 ))->setHelp('The subnet is used for determining the network that is tunnelled.');
 
-$section->addInput(new Form_Checkbox(
+$group->add(new Form_Checkbox(
 	'link1',
 	'Add Static Route',
 	'Add an explicit static route for the remote inner tunnel address/subnet via the local tunnel address',
 	$pconfig['link1']
 ));
 
+$section->add($group);
+
+$group = new Form_Group(gettext('IPv6'));
+
+$group->add(new Form_IpAddress(
+	'tunnel-local-addr6',
+	'*GRE tunnel local v6 address',
```

Comment:
Just making one comment but it applies to all the fields on this page. Since we're making changes, they should be in "Title Case" and the "GRE" in all of them is redundant (the whole page is GRE...)

So: "*Tunnel Local IPv6 Address". And the IPv4 versions should be changed to match, and explicitly labeled with IPv4, for example: "*Tunnel Local IPv4 Address"

Alternately, consider leading with the address family: "*IPv6 Tunnel Local Address" / "*IPv4 Tunnel Local Address"